### PR TITLE
Add splitdebt to DG

### PR DESCRIPTION
**Pull request by WR3nd3**

### CODE REVIEW

**Review by lyuting47 — APPROVED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -175,6 +175,28 @@ The activity diagrams below detail the behaviour of PayMeLah when a user inputs
...
+displayed list is '0'. It should also be noted that there is also never such a `Person`. The non-zero `Index` objects of the `Person` objects splitting the `Debt` are contained in a set `DeleteDebtCommandParser` creates.
+The `SplitDebtCommandParser` will construct a `SplitDebtCommand` object with this set of `Index` objects and the `Debt` object created.
+
+Consider a scenario where the user wishes to split a $30 food debt amongst multiple people. To speed up adding similar `Debt` objects to the `DebtList` of more than 1 `Person`, the `SplitDebtCommand` uses the set of `Index` objects. For each `Person` that an `Index` corresponds to, a new `Debt` object will be added to the `DebtList` of the specified `Person`.
+Some commands such as marking debts as paid, or possible future extensions such as editing a debt, may require modifying a `Debt` object in the `DebtList` of a `Person`. To ensure that modifying this `Debt` for 1 `Person` does not also erroneously modify the `Debt` of another `Person`, during execution of `splitdebt`, each `Debt` object is only added to one `DebtList`, and an `equal` instance of `Debt` is created and added for each other `DebtList`.
```
Comment (waynezsy) [inline docs/DeveloperGuide.md:192]: imo shorten this part since you are merely copying from adddebt

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -175,6 +175,28 @@ The activity diagrams below detail the behaviour of PayMeLah when a user inputs
...
+
+To enable the user to retroactively add a `Debt` that is backdated, the `SplitDebtCommandParser` can take in optional `<date>` and `<time>` parameters. By making these parameters optional, a default behaviour can be implemented such that when neither parameter is specified, a `Debt` object with the current date and time is created. This will improve the efficiency at which users can input new `Debt` objects for the (expected) most common scenario where they add the `Debt` into PayMeLah on the actual day the debt occurred.
```
Comment (waynezsy) [inline docs/DeveloperGuide.md:194]: this one is also copied, can shorten or remove since it does not differ from adddebt.

**Review by waynezsy — CHANGES_REQUESTED** [review]

**Review by waynezsy — APPROVED** [review]
LGTM

**Review by lyuting47 — APPROVED** [review]
LGTM

**Pull request merged**